Return error message if user tapped "Don't Allow"

extempl opened the pull request:
Based on https://stackoverflow.com/a/34189489/506954

macdonst commented:
@extempl I've been testing this but I can't get it to fail when the user clicks "Don't Allow" the code never seems to get called. Can you give me a better reproduction scenario?

extempl commented:
@macdonst That was a bit obsolete version, I've updated it to how it works on my end now:
Added two events `allowed` and `disallowed`. For checking access it uses same approach as used in `hasPermission` method. 
Also added ability to `off` all handlers of certain event, as `allowed` will fire each time app is resumed, so, on my end, I am listening for `allowed`/`disallowed` event only once before the first register. For `resume` I using built-in `hasPermissions` method.

Probably the code is need in polishing, I am not an ios-dev.

macdonst commented on src/ios/PushPlugin.m:
```diff
@@ -282,6 +282,28 @@ - (void)init:(CDVInvokedUrlCommand*)command;
+                   enabled = [appDelegate performSelector:@selector(userHasRemoteNotificationsEnabled)];
+               }
+               if (enabled) {
+                   NSLog(@"User allowed Notifications for the App");
```
I'm cool with the log but I think we can lose everything else in this if.

macdonst commented on www/push.js:
```diff
@@ -34,7 +34,9 @@ var PushNotification = function () {
       notification: [],
-      error: []
+      error: [],
+      allowed: [],
```
Let's get rid of the allowed/disallowed event listeners.

macdonst commented on www/push.js:
```diff
@@ -47,7 +49,9 @@ var PushNotification = function () {
     // triggered on registration and notification
     var success = function success(result) {
-      if (result && typeof result.registrationId !== 'undefined') {
+      if (result === 'allowed') {
```
Scratch this.

macdonst commented on src/ios/PushPlugin.m:
```diff
@@ -282,6 +282,28 @@ - (void)init:(CDVInvokedUrlCommand*)command;
+               }
+               else {
+                   NSLog(@"User disallowed Notifications for the App");
+                   [self failWithMessage:self.callbackId withMsg:@"disallowed" withError:Nil];
```
Maybe just give a more detailed error message here.

macdonst commented on www/push.js:
```diff
@@ -77,6 +81,9 @@ var PushNotification = function () {
 
     // triggered on error
     var fail = function fail(msg) {
+      if (msg === 'disallowed') {
```
Probably just get rid of this and let the regular error handler take care of it.
